packages/core/core/src/loaders/apis.ts

```diff
@@ -168,21 +172,22 @@ const loadDir = async (dir: string) => {
 
     const key = basename(fd.name, extname(fd.name));
 
-    root[normalizeName(key)] = await loadFile(join(dir, fd.name));
+    root[normalizeName(key)] = (await loadFile(join(dir, fd.name))).result;
   }
 
-  return root;
+  return { result: root };
 };
 
-const loadFile = (file: string) => {
+// because this is async and its contents are dynamic, we must return it as an array to avoid a property called `then` being interpreted as a Promise
+const loadFile = async (file: string): Promise<{ result: unknown }> => {
   const ext = extname(file);
 
   switch (ext) {
     case '.js':
-      return importDefault(file);
+      return { result: await importDefault(file) };
```

Comment:
importDefault is sync I believe

Reply:
Hahaha, so the whole problem could actually just be removing the await that is 10 lines above this. I will check on that.